Open Judge System/Web/OJS.Web/Infrastructure/Filters/ValidateRemoteDataApiKeyFilter.cs

```diff
@@ -31,8 +32,7 @@ public class ValidateRemoteDataApiKeyFilter : IActionFilter<ValidateRemoteDataAp
             var apiKey = request.QueryString[ApiKeyQueryStringParamName] ?? request[ApiKeyQueryStringParamName];
 
             var isValidApiKey = !string.IsNullOrWhiteSpace(apiKey) &&
-                this.userManager.Users.Any(u => u.Id == apiKey && u.IsDeleted == false) &&
-                this.userManager.IsInRole(apiKey, GlobalConstants.AdministratorRoleName);
+                                (this.IsApiKeyInAdminUsers(apiKey) || this.IsApiKeyInSettings(apiKey));
```

Comment:
Better swap the order of the `||` checks, to NOT make db call if apiKey is in Settings